### PR TITLE
feat: common labels and service monitor support

### LINKED ISSUE #125: add Service Monitor
Hi,

It would be a nice feature to add a serviceMonitor to this chart so that when rolling out this monitor if added.
The user would only be in charge afterward to add a secret, since we need a token from the UI to expose the `/metrics` I dont think it is possible to create the token beforehand.

I would be available to create a PR for this if you are open to adding a Service Montitor!

### LINKED ISSUE #126: Add CommonLabels
I use charts from bitnami quite a lot and one thing they have nailed down is adding a `commonLabels` value with labels that every resource uses, this could be a nice addition to this helm chart.

If this is somethng you like I could take the time to create a PR :D

### PR DESCRIPTION
**Pull request by vanhoutenbos**
I superseded #127 and #128 and promoted to release 1.1.0 + added a bug fix

I am fairly confident that this is it, this is the chart we forked privately and added some good comments from you on top.
Just not sure how to validate the chart besides linting it, I have no repo at hand for pushing it to and pulling the created image :(

### CODE REVIEW

**Comment by k3rnelpan1c-dev** [comment]
Hey, appreciate the effort!
Just one nit upfront, despite these being smaller additions, I prefer smaller PRs over one collection PR.
No issue here, just wanted to make the note :wink: 

In regards to testing this, I can validate it, however, the changes don't touch on critical functionality, so I don't see a too high risk of breaking anything.

**Comment by vanhoutenbos** [comment]
Not a problem at all!

Might have mis understood your comment but I thought you meant that I should bundle this. My apologies! https://github.com/k3rnelpan1c-dev/uptime-kuma-helm/pull/128#discussion_r1276785662

File: uptime-kuma/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: ServiceAccount
 metadata:
   name: {{ include "uptime-kuma.serviceAccountName" . }}
   labels:
-    {{- include "uptime-kuma.labels" . | nindent 4 }}
```
Comment (k3rnelpan1c-dev) [inline uptime-kuma/templates/serviceaccount.yaml]: while it is just for formatting, please keep the "trim whitespace/newlines" operator on the lavels

File: uptime-kuma/templates/tests/test-connection.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ kind: Pod
 metadata:
   name: "{{ include "uptime-kuma.fullname" . }}-test-connection"
   labels:
-    {{- include "uptime-kuma.labels" . | nindent 4 }}
```
Comment (k3rnelpan1c-dev) [inline uptime-kuma/templates/tests/test-connection.yaml]: same

File: uptime-kuma/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        key: password
+      username:
+        name: uptime-kuma-credentials
+        key: username
+  namespaceSelector: # Scrape metrics from...
+    matchNames:
+    - {{ include "uptime-kuma.serviceMonitorNamespace" . }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "uptime-kuma.fullname" . }}
```
Comment (k3rnelpan1c-dev) [inline uptime-kuma/templates/servicemonitor.yaml]: you seem to be adding an enable toggle to the `values` but it has no effect :wink: 
ideally we only provision the secret resource if the user opts to enable the monitor, WDYT?

*and please add the trim `-` in front of the label include*

**Comment by k3rnelpan1c-dev** [comment]
> Not a problem at all!
> 
> Might have mis understood your comment but I thought you meant that I should bundle this. My apologies! [#128 (comment)](https://github.com/k3rnelpan1c-dev/uptime-kuma-helm/pull/128#discussion_r1276785662)

ah :sweat_smile:, my bad, yeah no issue, like I said

**Review by k3rnelpan1c-dev — APPROVED** [review]
took the liberty to apply my feedback and fix the ADOC formatting

**Comment by k3rnelpan1c-dev** [comment]
Thank you for your contribution Jean-Paul!

**Comment by vanhoutenbos** [comment]
> took the liberty to apply my feedback and fix the ADOC formatting

Thank you! I've been crazy busy at work the last 2 weeks due to holidays!

**Pull request merged**